src/VisualStudio/Core/Def/Implementation/ProjectSystem/DocumentProvider.cs

```diff
@@ -25,19 +26,20 @@ namespace Microsoft.VisualStudio.LanguageServices.Implementation.ProjectSystem
     /// in that they watch the running document table, tracking open/close events, and also file
     /// change events while the file is closed.
     /// </summary>
-    internal abstract partial class DocumentProvider : ForegroundThreadAffinitizedObject
+    internal partial class DocumentProvider : ForegroundThreadAffinitizedObject
     {
-        #region Immutable readonly fields that can be accessed from foreground or background threads - do not need locking for access.
+        #region Immutable readonly fields/properties that can be accessed from foreground or background threads - do not need locking for access.
         private readonly object _gate = new object();
         private readonly uint _runningDocumentTableEventCookie;
         private readonly IVisualStudioHostProjectContainer _projectContainer;
         private readonly IVsFileChangeEx _fileChangeService;
         private readonly IVsTextManager _textManager;
         private readonly ITextUndoHistoryRegistry _textUndoHistoryRegistry;
-        protected readonly IVsRunningDocumentTable4 RunningDocumentTable;
-        protected readonly bool IsRoslynPackageInstalled;
-        protected readonly IVsEditorAdaptersFactoryService EditorAdaptersFactoryService;
-        protected readonly IContentTypeRegistryService ContentTypeRegistryService;
+        protected IVsRunningDocumentTable4 RunningDocumentTable { get; }
+        protected bool IsRoslynPackageInstalled { get; }
+        protected IVsEditorAdaptersFactoryService EditorAdaptersFactoryService { get; }
+        protected IContentTypeRegistryService ContentTypeRegistryService { get; }
+        protected VisualStudioDocumentTrackingService DocumentTrackingService { get; }
```

Comment:
I'd look through source control to understand when it was added. The tracking is being used to start/stop the solution crawler, so perhaps the belief is MiscellaneousFilesWorkspace doesn't care, or we only need one watching to track the singleton crawler?

I don't see why we wouldn't just do that as a part of this PR instead of deferring work.